pynance/textimporter.py

```diff
@@ -108,6 +146,56 @@ def __init__(self,
         self.formatters = formatters
         self.skiprows = skiprows
         self.encoding = encoding
+        self.total_balance_re_pattern = total_balance_re_pattern
+
+    def read_total_balance(self, filepath_or_buffer):
+        """
+        Parses a filepath or buffer for a regex given in
+        total_balance_re_pattern and formats it into a number
+
+        PARAMS:
+        -------
+        filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or
+            any object with a readlines() method
+
+        RETURNS:
+        --------
+        number :
+            total balance as specified of the type that is defined for the
+            `total_balance` column using a formatter given in the
+            formatters dict
+
+        RAISES:
+        -------
+        UnsupportedCsvFormat :
+            if the total_balance was not found in the given filepath or buffer
+
+        """
+
+        # TODO: check if we need pandas.io.common.get_filepath_or_buffer
```

Comment:
Do we really plan to do this? If so, could you please add a short issue? In any case, in my experience `TODO`s in code tend to stick around for very lone :) Maybe we should remove it?

Reply:
yes, this was more a note to myself. When I added the part where the header is searched for the regex pattern, I wanted to maintain the full capabilities that pandas offers in the `read_csv` - I think it's pretty clear where to look at in the pandas code if we want to mimic that behaviour and the note can be removed